include/alpaka/kernel/TaskKernelCpuThreads.hpp

```diff
@@ -76,137 +77,96 @@ namespace alpaka
             Vec<TDim, TIdx> const& blockThreadExtent,
             ThreadPool& threadPool,
             TKernelFnObj const& kernelFnObj,
             std::decay_t<TArgs> const&... args) -> void
         {
-            // The futures of the threads in the current block.
             std::vector<std::future<void>> futuresInBlock;
-
-            // Set the index of the current block
             acc.m_gridBlockIdx = gridBlockIdx;
 
-            // Bind the kernel and its arguments to the host block thread execution function.
-            auto boundBlockThreadExecHost = std::bind(
-                &TaskKernelCpuThreads::blockThreadExecHost,
-                std::ref(acc),
-                std::ref(futuresInBlock),
-                std::placeholders::_1,
-                std::ref(threadPool),
-                std::ref(kernelFnObj),
-                std::ref(args)...);
-            // Execute the block threads in parallel.
-            meta::ndLoopIncIdx(blockThreadExtent, boundBlockThreadExecHost);
+            // Execute the threads of this block in parallel.
+            meta::ndLoopIncIdx(
+                blockThreadExtent,
+                [&](Vec<TDim, TIdx> const& blockThreadIdx)
+                {
+                    // copy blockThreadIdx because it will get changed for the next iteration/thread.
+                    futuresInBlock.emplace_back(threadPool.enqueueTask(
+                        [&, blockThreadIdx] { runThread(acc, blockThreadIdx, kernelFnObj, args...); }));
+                });
+
             // Wait for the completion of the block thread kernels.
             for(auto& t : futuresInBlock)
                 t.wait();
+
             // Clean up.
             futuresInBlock.clear();
-
             acc.m_threadToIndexMap.clear();
-
-            // After a block has been processed, the shared memory has to be deleted.
-            freeSharedVars(acc);
-        }
-        //! The function executed for each block thread on the host.
-        ALPAKA_FN_HOST static auto blockThreadExecHost(
-            AccCpuThreads<TDim, TIdx>& acc,
-            std::vector<std::future<void>>& futuresInBlock,
-            Vec<TDim, TIdx> const& blockThreadIdx,
-            ThreadPool& threadPool,
-            TKernelFnObj const& kernelFnObj,
-            std::decay_t<TArgs> const&... args) -> void
-        {
-            // Bind the arguments to the accelerator block thread execution function.
-            // The blockThreadIdx is required to be copied in because the variable will get changed for the next
-            // iteration/thread.
-            auto boundBlockThreadExecAcc
-                = [&, blockThreadIdx]() { blockThreadExecAcc(acc, blockThreadIdx, kernelFnObj, args...); };
-            // Add the bound function to the block thread pool.
-            futuresInBlock.emplace_back(threadPool.enqueueTask(boundBlockThreadExecAcc));
+            freeSharedVars(acc); // After a block has been processed, the shared memory has to be deleted.
```

Comment:
I do not know why this comment is now behind the line. This could end up depending on the indention that the comment is broken into two lines and then looks very strange.

Reply:
I put it there to save a LOC :)